pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh

```diff
@@ -150,16 +150,16 @@ convertDesktopFile() {
     local -r file=$1
     local -r sharePath=$(dirname "$(dirname "$file")")
     local -r name=$(getDesktopParam "${file}" "^Name")
-    local -r exec=$(getDesktopParam "${file}" "Exec")
+    local -r exec=$(getDesktopParam "${file}" "Exec" | sed -e 's/ %[fFuU]$//')
```

Comment:
why this sed?

Reply:
It's in the commit message:
> The "Exec" key in desktop items sometimes has one of the `%f`, `%F`,
`%u` and `%U` suffixes, which specify whether the command takes a file,
multiple files or a generalized URL or URLs. Darwin application bundles
do no understand this syntax so we do the next best thing, which is
simply dropping it.

Reply:
Yeah, that may be a good idea.
Slightly tangent to the work here though, which is about getting it to actually work when first : )

Reply:
What would I use to log from a setup hook?

Reply:
echo : )

something like

```
    ${if customExec != null then ''
      local -r exec=${lib.escapeShellArg customExec}
    '' else ''
      local -r execRaw=$(getDesktopParam "${file}" "Exec")
      local -r exec=$(echo "$execRaw" | sed -e 's/ %[fFuU]$//')
      [[ "$exec" != "$execRaw" ]] && echo "makeDarwinBundle: patching exec from '$execRaw' to '$exec'. maybe override with customExec"
    ''}
```

Reply:
I added a new custom desktop entry key, like for the icon squircling. It has to be entirely shell since this isn't in a Nix expression.